src/components/Swap/Swap.tsx

```diff
@@ -1,70 +1,88 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faExchangeAlt } from '@fortawesome/free-solid-svg-icons';
 import {
   AssetAmount,
-  AssetInfo,
   BoxSelection,
   DefaultBoxSelector,
   DefaultTxAssembler,
 } from 'ergo-dex-sdk/build/module/ergo';
 import { fromAddress } from 'ergo-dex-sdk/build/module/ergo/entities/publicKey';
-import { useGetAllPools } from '../../hooks/useGetAllPools';
-import { getTokenInfo } from '../../utils/getTokenInfo';
+import { YoroiProver } from '../../utils/yoroiProver';
 import { WalletContext } from '../../context/WalletContext';
+import { useGetAllPools } from '../../hooks/useGetAllPools';
+import { PoolSelect } from '../PoolSelect/PoolSelect';
 import { defaultMinerFee } from '../../constants/erg';
 import { getButtonState, WalletStates } from './utils';
 
-interface Swap {
-  isWalletConnected: boolean;
+interface SwapFormProps {
+  pools: AmmPool[];
 }
 
-export const Swap = () => {
+const SwapForm: React.FC<SwapFormProps> = ({ pools }) => {
   const { isWalletConnected } = useContext(WalletContext);
+  const [selectedPool, setSelectedPool] = useState<AmmPool | undefined>();
+  const [inputAssetAmount, setInputAssetAmount] =
+    useState<AssetAmount | undefined>();
+  const [outputAssetAmount, setOutputAssetAmount] =
+    useState<AssetAmount | undefined>();
+  const [inputAmount, setInputAmount] = useState('');
+  const [outputAmount, setOutputAmount] = useState('');
+  const [selectedAddress, setSelectedAddress] = useState('');
   const [feePerToken, setFeePerToken] = useState('');
-  const [firstTokenId, setFirstTokenId] = useState('');
-  const [secondTokenId, setSecondTokenId] = useState('');
-  const [firstTokenAmount, setFirstTokenAmount] = useState('');
-  const [secondTokenAmount, setSecondTokenAmount] = useState('');
-  const [firstTokenInfo, setFirstTokenInfo] = useState<AssetInfo | undefined>(
-    undefined,
-  );
-  const [secondTokenInfo, setSecondTokenInfo] = useState<AssetInfo | undefined>(
-    undefined,
-  );
-  const pools = useGetAllPools();
 
-  const choosedPool = useMemo(() => {
-    if (!firstTokenId.trim() || !secondTokenId.trim()) {
-      return null;
+  useEffect(() => {
+    if (selectedPool === undefined) {
+      updateSelectedPool(pools[0]);
     }
-    return pools?.find(
-      (ammPool) =>
-        (ammPool.assetX.id === firstTokenId &&
-          ammPool.assetY.id === secondTokenId) ||
-        (ammPool.assetX.id === secondTokenId &&
-          ammPool.assetY.id === firstTokenId),
-    );
-  }, [firstTokenId, secondTokenId, pools]);
-  const [chooseAddress, setChooseAddress] = useState('');
+  }, [pools]);
+
+  const updateSelectedPool = useCallback((pool: AmmPool) => {
+    setSelectedPool(pool);
+    setInputAssetAmount(pool.x);
+    setOutputAssetAmount(pool.y);
+  }, []);
+
+  const handleSwitchAssets = useCallback(() => {
+    console.log('hehh');
+    setInputAssetAmount(outputAssetAmount);
+    setOutputAssetAmount(inputAssetAmount);
```

Comment:
after user click to "switch button" you need use again .inputAmount function from AMMPool to recalculate output input.
Scenario: 
1. User have 1 in input amount and 2 in output amount (result from inputAmount function from AmmPool)
2. User click to 'switch button'
3. Now you need write in input 2 and in output result from inputAmount function from AmmPool.